### ISSUE TITLE
[MMaps] Pet/Charge Pathing in Arena

### ISSUE THREAD

**Issue by Xarper** [TrinityCore/TrinityCore#7592]
Blade's Edge Arena is the main issue here, on other arenas (Ruins/Nagrand) the pet and charge pathing seems to work really well as long as pet.vmaplos is disabled.

1. The ropes that go on top of the pillars in Blade's Edge dont seem to have any "pathing" on them (Pets wont cross them if set to attack and Warriors cant charge anyone on top of the pillars)
If Pet is on follow it will fly across the air to follow the master and usually fall down in the progress.

2. Charge only calculates the most straightforward distance from you to the target instead of calculating the actual distance of the charge path, this means you can charge someone on top of the bridge from below the bridge and have insanely long charge pathing
Easy fix would be to just implement charge to check the distance of the actual path calculated before executing it?

3. Charge Pathing is disabled by default in the files? If you don't enable it the Warrior will just fly through the air just like it does without MMaps.
In MotionMaster.h:
void MoveCharge(float x, float y, float z, float speed = SPEED_CHARGE, uint32 id = EVENT_CHARGE, bool generatePath = false);

**Comment by Raydor** [comment]
Yeah still no points on the ropes and also, if you charge onto anyone who is right in the edge, you charge into him without following the MMAP Points (in the photos I use .mmap path to reproduce the issue better):

http://puu.sh/e0AVB/a1f07186e7.jpg

http://puu.sh/e0AW1/18828dbecd.jpg

http://puu.sh/e0AY9/c06bb86df1.jpg

http://puu.sh/e0AYt/f5ae1d1d32.jpg

**Comment by chaodhib** [comment]
The ropes that link the bridge with the pillars (where the crystals pop) are still unusable by NPCs so the issue should be reopened.

didn't test the charge thing however.

tested on d7600f1126d2780fecdfcdfc66a598a48cee0136

**Comment by Nawuko** [comment]
Just generate some offmesh for the ropes = fixed.

**Comment by chaodhib** [comment]
and how does one do that?

**Comment by Nawuko** [comment]
my english is to bad to explain in a good way. Ask @jackpoz he knows how :)

**Comment by Nawuko** [comment]
Read the propper command here: https://github.com/TrinityCore/TrinityCore/tree/6.x/src/tools/mmaps_generator/Info

**Comment by jackpoz** [comment]
> Just generate some offmesh for the ropes = fixed.

not really since they'll be jump connections and the pet will get stuck jumping back and forward from the rope

**Comment by Nawuko** [comment]
Offmeshs are working fine for me, no pet issue at all. Fixes also warrior charge on the platform.

Oh yea and i was only reffering to "The ropes that link the bridge with the pillars (where the crystals pop) are still unusable by NPCs so the issue should be reopened." not to the issue overall :)

**Comment by jackpoz** [comment]
then you probably didn't test them enough to find the issue

**Comment by Nawuko** [comment]
I'm very certain that this issue would have came up, using the offmesh since over half a year without any reports of broken pathing for the pillars

**Comment by jackpoz** [comment]
no clue, you didn't post your offmesh connections yet

**Comment by chaodhib** [comment]
@jackpoz "The real fix would be to change the rope m"esh to be wide enough to be walkable." How can we do that? Aren't the mesh auto generated?

**Comment by Xaxoxuxu** [comment]
how can one create an offmesh ?

**Comment by Aokromes** [comment]
This bug is still valid after fe68b2e99851a888c5182d7bebddfea67459afd3 ?

**Comment by jackpoz** [comment]
yes

**Comment by Goatform** [comment]
Yeah, it is still broken.

**Comment by Keader** [comment]
@nawuko can you share your offmesh?

**Comment by kvipka** [comment]
http://rustemu.org/2018/04/10/fixed-blade-edge-arena/

**Comment by funjoker** [comment]
@kvipka that does not help anybody here.

**Comment by Keader** [comment]
@jackpoz when you have time can try find a offmesh for it?
